rust_src/src/frames.rs

```diff
@@ -534,4 +534,14 @@ pub fn frame_after_make_frame(frame: LispFrameOrSelected, made: LispObject) -> L
     made
 }
 
+/// Return the frame to which FRAME's keystrokes are currently being sent.
+/// If FRAME is omitted or nil, the selected frame is used.
+/// Return nil if FRAME's focus is not redirected.
+/// See `redirect-frame-focus'.
+#[lisp_fn(min = "0")]
+pub fn frame_focus(frame: LispFrameOrSelected) -> LispObject {
+    let frame_ref = frame.live_or_error();
```

Comment:
It would be preferred if you used `LispFrameLiveOrSelected` here instead, and then did
```rust
let frame_ref: LispFrameRef = frame.into()
```